stdlib/experimental/experimental.flux

```diff
@@ -1320,3 +1320,51 @@ builtin unpivot : (<-tables: stream[{A with _time: time}]) => stream[{B with _fi
 // ## Metadata
 // introduced: 0.174.0
 builtin catch : (fn: () => A) => string
+
+// diff takes two table streams as input and produces a diff.
+//
+// `experimental.diff()` compares tables with the same group key.
+// If compared tables are different, the function returns a table for that group key with one or more rows.
+// If there are no differences, the function does not return a table for that group key.
+//
+// **Note:** `experimental.diff()` cannot tell the difference between an empty table and a non-existent table.
+//
+// **Important:** The output format of the diff is not considered stable and the algorithm used to produce the diff may change.
+// The only guarantees are those mentioned above.
+//
+// ## Parameters
+// - want: Input stream for the `-` side of the diff.
+// - got: Input stream for the `+` side of the diff.
+// - epsilon: Specify how far apart two float values can be, but still be considered equal. Default is `0.000000001`.
+// - nansEqual: Consider `NaN` float values as equal. Default is `false`.
```

Comment:
I'd personally say no. I'd even remove epsilon and just consider some default.

My personal reason for removing epsilon is that future optimization of this probably depends on hashing the inputs and doing hash comparisons instead of the current column comparisons. Hash comparisons work for everything except floats where epsilon has to be considered. If we expose an epsilon option, we also have to ensure that whatever we do in the future works with epsilon, but if epsilon is just kind of hidden, we can potentially do something equivalent that doesn't involve a specific implementation.

Now I don't know if that exists. I just know that adding `epsilon` as a parameter leaks an implementation detail.

If we're fine with that, I'll remove both of these or I can just remove one.

Reply:
I wonder if it makes sense to expose `epsilon` as an `option` in the `math` package?

Reply:
I was thinking that using `fmt.Sprintf("%.3f", f)` or something might work? It would output the float up to three decimal places which is kind of the equivalent of what epsilon is supposed to do. I'm not sure if that works though and I don't really want to specify how a solution would work. Bit by bit definitely doesn't work though.

I think that's part of my reason for wanting to remove `epsilon`. I think the above wouldn't work if we had `epsilon` as a parameter because I could theoretically put any value there as the epsilon and it might not be possible with the above method. Once we figure it out, we can add appropriate parameters if needed.

Reply:
I heard somewhere that some Intel processors use 80-bit registers to do math, but still store them as 64-bit values, so the compiler generating a `store` and a `load` might result in two values comparing as different, even if they would otherwise be identical. If this is the case, we'd be entirely at the mercy of the foibles of the compiler and architecture---seems really hard to work with portably?